lib/rubygems/ext/builder.rb

```diff
@@ -88,13 +88,20 @@ def self.run(command, results, command_name = nil, dir = Dir.pwd, env = {})
       # Set $SOURCE_DATE_EPOCH for the subprocess.
       build_env = { "SOURCE_DATE_EPOCH" => Gem.source_date_epoch_string }.merge(env)
       output, status = begin
-                         Open3.capture2e(build_env, *command, chdir: dir)
+                         Open3.popen2e(build_env, *command, chdir: dir) do |_stdin, stdouterr, wait_thread|
+                           output = String.new
+                           while line = stdouterr.gets
+                             output << line
+                             if verbose
+                               print line
+                             end
+                           end
+                           [output, wait_thread.value]
```

Comment:
Since `results` is in scope here would this make sense?

```suggestion
                           while line = stdouterr.gets
                             results << line
                             if verbose
                               print line
                             end
                           end
                           wait_thread.value
```

Reply:
Since `results` is an Array, this will make a change in structure (`["output\noutput\noutput\n"]` vs `["output\n", "output\n", "output\n"]`).

While there might not be much practical impact, I'm not sure removing a String allocation is worth changing `#build`'s result. What do you think?

Reply:
Ah, ok. I didn't look high enough up the method. That makes sense then.